packages/engine-core/src/common/Engine.ts

```diff
@@ -23,7 +19,17 @@ export type BatchTransactionOptions = {
   isolationLevel?: Transaction.IsolationLevel
 }
 
-export type InteractiveTransactionOptions<Payload> = Transaction.Info<Payload>
+export type TransactionOptions<InteractiveTransactionPayload> =
```

Comment:
Do you think it would be possible to re-use `PrismaPromiseBatchTransaction` and `PrismaPromiseInteractiveTransaction`?

Reply:
I was thinking about it, but in the end decided to not do it - batch id does not have any meaning to the engine, it's purely client invention, so it felt wrong adding it to the engine types, but we absolutely need it in the client.